src/alertsmanagement/azext_alertsmanagement/custom.py

```diff
@@ -0,0 +1,345 @@
+                                      target_resource_type=None,
+                                      target_resource=None,
+                                      severity=None,
+                                      monitor_service=None,
+                                      impacted_scope=None,
+                                      description=None,
+                                      alert_rule_id=None,
+                                      action_group=None,
+                                      name=None):
+    if resource_group_name is not None:
+        return client.list_by_resource_group(
+            resource_group_name=resource_group_name,
+            target_resource_group=target_resource_group,
+            target_resource_type=target_resource_type,
+            target_resource=target_resource,
+            severity=severity,
+            monitor_service=monitor_service,
+            impacted_scope=impacted_scope,
+            description=description,
+            alert_rule_id=alert_rule_id,
+            action_group=action_group,
+            name=name)
+    return client.list_by_subscription(
+        target_resource_group=target_resource_group,
+        target_resource_type=target_resource_type,
+        target_resource=target_resource,
+        severity=severity,
+        monitor_service=monitor_service,
+        impacted_scope=impacted_scope,
+        description=description,
+        alert_rule_id=alert_rule_id,
+        action_group=action_group,
+        name=name)
+
+
+def create_alertsmanagement_smart_detector_alert_rule(cmd, client,
+                                                      resource_group,
+                                                      name,
+                                                      location=None,
+                                                      tags=None,
+                                                      description=None,
+                                                      state=None,
+                                                      severity=None,
+                                                      frequency=None,
+                                                      action_groups=None,
+                                                      throttling=None,
+                                                      detector=None,
+                                                      scope=None):
+    body = {}
+    body['location'] = location  # str
+    body['tags'] = tags  # unknown-primary[object]
+    body['description'] = description  # str
+    body['state'] = state  # str
+    body['severity'] = severity  # str
+    body['frequency'] = frequency  # unknown-primary[timeSpan]
+    # body['action_groups'] = json.loads(action_groups) if isinstance(action_groups, str) else action_groups
+    body['action_groups'] = {'groupIds': action_groups}
+    body['throttling'] = json.loads(throttling) if isinstance(throttling, str) else throttling
```

Comment:
you can use `type=shell_safe_json_parse` in _params.py, then you don't need this extra handling here.